[tvOS] Login Flow Cleanup - Second Pass

JPKribs opened the pull request:
# Login Flow UI Updates

## Initial Login Screen
<details>
<summary>Connect to Server - Remove Duplicate Icons</summary>

**New Version:**
![New Initial Screen](https://github.com/user-attachments/assets/b3c38f41-4e64-43c2-8c52-ad02342c62ee)

**Previous Version:**
![Old Initial Screen](https://github.com/user-attachments/assets/32cb7fc6-6c15-4376-8dc5-3bfb7a1eaf35)
</details>

## Server Management
<details>
<summary>Server Bar - Ensure Menus are same sizing</summary>

**New Version:**
![New Server Bar](https://github.com/user-attachments/assets/688ce730-a50a-47ee-a8db-fcecf618018a)

**Previous Version:**
![Old Server Bar](https://github.com/user-attachments/assets/46a0d8a0-08f6-408d-9b62-317e991c4c3d)
</details>

## Server Configuration
<details>
<summary>Edit Server View - Clean up menu buttons</summary>

![New Edit Server View](https://github.com/user-attachments/assets/195bd319-291b-4afb-ab08-c2d42d4bbde4)
</details>

## Additional UI Elements
<details>
<summary>User Sign-In - Add Background Image for Server</summary>

![Login View Background](https://github.com/user-attachments/assets/143cac31-aa68-448d-bff1-d2db74b0cb1e)
</details>

---

## Outstanding Issue
<details>
<summary>tvOS User Sign-In Crash</summary>

This could be an emulator issue but I see this issue where the first time I login with a new user, Swiftfin crashes. Restarting, the user still exists but it's not great to restart after each login.

<img width="405" alt="Screenshot 2025-01-21 at 07 52 19" src="https://github.com/user-attachments/assets/17614b4e-3b04-45ee-806c-b2ed76e31c88" />

LePips commented on Swiftfin tvOS/Views/SelectUserView/Components/UserGridButton.swift:
```diff
@@ -71,7 +71,13 @@ extension SelectUserView {
             .overlay {
-                if isEditing {
+                if isEditing && isSelected {
+                    RelativeSystemImageView(systemName: "trash.circle.fill", ratio: 1)
```
I would prefer if this was similar to iOS with a smaller checkmark icon in the `bottomTrailing` position.

JPKribs replied:
I have a version of this. I am doing the hoverEffect on both the checkbox AND the icon since this is what this looks like without it:

https://github.com/user-attachments/assets/9aa8920e-18a5-4ca9-8d93-6e00295f7d5a

LePips replied:
Hm, I didn't realize that the `buttonBorderShape(.circle)` would cause issues with this overlay. Instead of `bottomTrailing`, I'm fine with just setting in the `.bottom` for now. It won't need a separate `hoverEffect`.

LePips reviewed (changes requested):
As I was typing a comment for [this change](https://github.com/jellyfin/Swiftfin/pull/1403/files#diff-21c9dad0b010a44f596130cb19fceb0151b5b104084312d6d9f38e05fd9e3da3R244) I thought instead to just implement the same `Menu/Button` conditioning for server selection when it is `all`.

Using the user selection a bit more, I think we may want to move the `Add User` button from an inline item with the users and instead have it be a button on the left/leading side of the server selection to make the stack symmetrical. Maybe also for iOS to move it to the top trailing menu. We can take a look at that later.

JPKribs commented:
> Using the user selection a bit more, I think we may want to move the `Add User` button from an inline item with the users and instead have it be a button on the left/leading side of the server selection to make the stack symmetrical. Maybe also for iOS to move it to the top trailing menu. We can take a look at that later.

I can do that! One item of note, on user sign in on tvOS, disabled buttons square off. IMO, it's a little wonky looking but not an issue per-say. Wanted to run it by you while I was touching up login flow:
![Simulator Screenshot - Apple TV 4K (3rd generation) - 2025-01-21 at 12 15 52](https://github.com/user-attachments/assets/2b839b96-9447-4dce-adc4-268230682a5d)

JPKribs commented:
> I think we may want to move the `Add User` button from an inline item with the users and instead have it be a button on the left/leading side of the server selection

This is my implementation of this. Instead of disabling/hiding when there is no selected server I turned this into a menu. 1) It's hard to mirror Button & Menu styling to this avoids that that headache. 2) you can now always select the server you want to add a user to regardless of if you are in 'All Servers' or have a selected server.

![Simulator Screenshot - Apple TV 4K (3rd generation) - 2025-01-21 at 13 08 10](https://github.com/user-attachments/assets/46260dc5-5f6c-4be7-a663-fefad491d323)

JPKribs commented:
Okay! Take 2. I've made some additional changes. 

- I updated the `ListRowButton` to have a `.destructive` that mirrors iOS.
- I updated the `ListRowButton` to retain its rounded corners even when disabled.
- I changed changed the add user button to be a menu to allow selecting the server you want to add a user from.
- There is only an add user, user card if there are no users. This is mostly to be a bigger target to let the user know where to add profiles.
- Delete button now uses the same `ListRowButton` `.destructive` to mirror.

<details>
<summary>Delete User Buttons</summary>

Selected User - Non-Focused Delete:
![Selected User   Non-Highlighted Delete](https://github.com/user-attachments/assets/326b2352-d3be-4a71-81a3-349193428894)

Selected User - Focused Delete:
![Selected User   Highlighted Delete](https://github.com/user-attachments/assets/4a362200-6d78-4c52-b319-a062c46e8272)

Disabled Delete:
![Disabled Delete](https://github.com/user-attachments/assets/17a13b2a-4a9c-4880-8e8d-f708b856b372)

</details>

<details>
<summary>Add User Buttons</summary>

This is the new layout. See - no 'Add User' button in the grid:
![Add User - Layout](https://github.com/user-attachments/assets/311c69a2-c69e-420a-a6c5-037ea5117898)

Adding a user. See - Choosing the server for adding a user:
![Add User - Some Existing User(s)](https://github.com/user-attachments/assets/64cb04c3-bc49-44ee-9882-131d8e7e5113)

Adding a user. See - The Add User Grid button exists
![Add User - No Existing Users](https://github.com/user-attachments/assets/cbd58aec-7c54-43d2-bc6d-1edf1f440ea0)

</details>


<details>
<summary>New Found Bug</summary>
Old Buttons were cards inside of forms... I actually did this a while ago so that's on me:
<img width="1362" alt="Screenshot 2025-01-21 at 15 03 16" src="https://github.com/user-attachments/assets/28d0804d-3491-4df0-88bd-eb0432b3a6d6" />

The new buttons fill this out appropriately:
<img width="1409" alt="Screenshot 2025-01-21 at 15 05 46" src="https://github.com/user-attachments/assets/da2b1dbc-f2f5-4ea0-9a1b-6ba8578bb349" />

<details>

---

<details>
<summary>New Found Bug</summary>

Single User Bug - Only occurs if you only have a single account. I believe this precedes my changes but before there was almost always 2 users since 1 user + the add user "user":
![Weird Bug](https://github.com/user-attachments/assets/44d62633-8591-4630-a289-56fc70fcce5b)

</details>

LePips reviewed (changes requested):
Let's leave out the add user button changes as that change would also consist of more cleanup within `SelectUserView`.

JPKribs commented:
> Let's leave out the add user button changes as that change would also consist of more cleanup within `SelectUserView`.

Sounds good. Those changes should now be reverted!

LePips commented on Swiftfin tvOS/Views/ServerDetailView.swift:
```diff
@@ -75,14 +75,17 @@ struct EditServerView: View {
                         }
                     }
+                    .listRowBackground(Color.clear)
+                    .listRowInsets(EdgeInsets(top: 0, leading: 0, bottom: 0, trailing: 0))
```
```suggestion
                    .listRowInsets(.zero)
```

LePips commented on Swiftfin tvOS/Views/ServerDetailView.swift:
```diff
@@ -75,14 +75,17 @@ struct EditServerView: View {
                         }
-                        .foregroundStyle(.primary, .red.opacity(0.5))
+                        .listRowBackground(Color.clear)
+                        .listRowInsets(EdgeInsets(top: 0, leading: 0, bottom: 0, trailing: 0))
```
```suggestion
                        .listRowInsets(.zero)
```

LePips commented on Translations/en.lproj/Localizable.strings:
```diff
@@ -1618,6 +1618,9 @@
 "selectImage" = "Select Image";
 
+/// Select Server
+"selectServer" = "Select Server";
```
```suggestion
"selectServer" = "Select server";
```

LePips reviewed (approved):
Looks great!
